### PR TITLE
PROTOCOL: Remove program manager

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
This removes the 'program manager' functionality that was added a while
back in an effort to buffer larger chunks of gcode.  Unfortunately, this
has put us in a bit of a bind and caused a lot of headaches related to
gcode failing to execute and has introduced a number of subtle race
conditions with our new cutting system.  By switching back, it makes
everything a bit more synchronous as we only get positive
confirmation ('ok') when a line has been added to the line buffer, NOT
when it has been added to the internal data buffer.

### CODE REVIEW

File: protocol.c [hunk 1]
```diff
@@ -287,7 +294,6 @@ void protocol_execute_runtime()
           sys.flags &= ~SYSFLAG_AUTOSTART; // Reset auto start per settings.
         }
       }
-      bit_false(SYS_EXEC,EXEC_CYCLE_START);
```
Comment (Jeff-Ciesielski) [inline protocol.c:299]: This prevents getting stuck in a queued state.

**Comment by knitHacker** [comment]
Okay no glaring errors I could see not knowing grbl that well.

File: protocol.c [hunk 2]
```diff
@@ -68,7 +67,7 @@ static uint8_t protocol_execute_line(char *line)
   }
 
   /* If there was an error, report it */
-  if (status && status != STATUS_IDLE_WAIT) {
+  if (status != STATUS_IDLE_WAIT) {
```
Comment (StephanCoetzee) [inline protocol.c:70]: Why did you change this? Can't status be null?

**Review by StephanCoetzee — COMMENTED** [review]
LGTM

File: protocol.c (hunk 2 above)
Comment (Jeff-Ciesielski) [reply protocol.c:70]: 'NULL' status would be OK, in which case we still wanted to go ahead and report it (which is how it was before, but I just want to avoid IDLE status)

Comment (StephanCoetzee) [reply protocol.c:70]: :+1:
